src/lwc/rd2EntryForm/rd2EntryForm.js

```diff
@@ -0,0 +1,318 @@
+
+import getSetting from '@salesforce/apex/RD2_entryFormController.getSetting';
+
+export default class rd2EntryForm extends LightningElement {
+
+    @api parentId;
+    @api recordId;
+    listenerEvent = 'rd2EntryFormEvent';
+
+    customLabels = Object.freeze({
+        cancelButtonLabel,
+        saveButtonLabel,
+        donationSectionHeader,
+        donorSectionHeader,
+        otherSectionHeader
+    });
+
+    @track fields = {};
+    @track record;
+    recurringDonationInfo;
+    fieldInfos;
+
+    @track header = newHeaderLabel;
+
+    @track isAutoNamingEnabled;
+    @track isMultiCurrencyEnabled;
+
+    @track contactId;
+    @track accountId;
+
+    @track isLoading = true;
+    @track isRecordReady = false;
+    isSettingReady = false;
+
+    @track hasError = false;
+    @track errorMessage = {};
+
+    /*******************************************************************************
+    * @description Dynamic render edit form CSS to show/hide the edit form 
+    */
+    get cssEditForm() {
+        return (!this.isLoading && this.isSettingReady && this.isRecordReady)
+            ? ''
+            : 'slds-hide';
+    }
+
+    /*******************************************************************************
+    * @description Get Recurring Donation SObject API name 
+    */
+    get recurringDonationName() {
+        return getNestedProperty(RECURRING_DONATION_INFO, 'objectApiName');
+    }
+
+
+    /*******************************************************************************
+    * @description Retrieve Recurring Donation Object info
+    */
+    @wire(getObjectInfo, {objectApiName: '$recurringDonationName' })
+    wiredRDObjectInfo(response) {
+        if (response.data) {
+            this.recurringDonationInfo = response.data;
+            this.buildDisplayedFields(this.recurringDonationInfo.fields);
+            this.fieldInfos = this.buildFieldDescribesForWiredMethod(
+                this.recurringDonationInfo.fields,
+                this.recurringDonationInfo.apiName);
+
+            if (this.recordId == null) {
+                this.isRecordReady = true;
+            }
+        }
+
+        if (response.error) {
+            this.isRecordReady = true;
+            handleError(response.error);
+        }
+    }
+
+    /*******************************************************************************
+    * @description Method converts field describe info into objects that the
+    * getRecord method can accept into its 'fields' parameter.
+    */
+    buildFieldDescribesForWiredMethod(fields, objectApiName) {
+        return Object.keys(fields).map((fieldApiName) => {
+            return {
+                fieldApiName: fieldApiName,
+                objectApiName: objectApiName
+            }
+        });
+    }
+
+    /*******************************************************************************
+    * @description Contrcut any needed field info from the Sobject Info
```

Comment:
`@description Construct field describe info from the Recurring Donation SObject info`